cmd/deployment_deleter.go

```diff
@@ -100,6 +100,14 @@ func (c *deploymentDeleter) DeleteDeployment(stage biui.Stage) (err error) {
 		return bosherr.WrapError(err, "Loading deployment state")
 	}
 
+	if deploymentState.CurrentIP != "" {
```

Comment:
I added a check in [vm.go](https://github.com/zhanggbj/bosh-cli/blob/a3b247d4d946a071bc280ef0d1383a419ca8096d/deployment/vm/vm.go#L216) to ensure that If current_ip is found, it won’t delete vm , so bosh-softlayer-cpi could do os-reload after. When command `delete-env` is issued, this line ensure current_ip is "", so director is always deleted even it has been os-reloaded before(current_ip in not "").